app/controllers/lists_controller.rb

```diff
@@ -0,0 +1,61 @@
+class ListsController < ApplicationController
+  before_action :require_signed_in!
+
+  def index
+    @lists = List.all
+  end
+
+  def show
+    @list = List.includes(:tasks).find(params[:id])
+    if !check_owner(@list)
+      redirect_to root_path
+    end
+    redirect_to list_tasks_url(@list)
+  end
+
+  def new
+    @list = List.new
+  end
+
+  def create
+    @list = List.new(list_params)
+    @list.user_id = current_user.id
+    if @list.save
+      redirect_to list_url(@list)
+    else
+      flash.now[:errors] = @list.errors.full_messages
```

Comment:
Am I correct that when we `render :new` the form itself will show the validation errors stored in `@list`? If so, is it useful to show the same errors in the flash as well?

Reply:
I believe the validation errors would have to rendered using an iteration ie: `<% flash.each do |key, value| %> <div><%= value %></div>` or similar methods in the html view file. In the case of the given tasks controller this line is given `format.json { render json: @task.errors, status: :unprocessable_entity }` which would commit it to a json object. The `_form.html.haml` then checks for the existence of those errors in the json object and renders them through an iteration. The method I used is similar except it does not create a json object and rather makes it accessible through `flash`.

Reply:
The `_form.html.haml` has:
```
  -if @list.errors.any?
    #error_explanation
      %h2="#{pluralize(@list.errors.count, "error")} prohibited this list from being saved:"
      %ul
        -@list.errors.full_messages.each do |msg|
          %li= msg
```

I'm saying that the error messages are displayed twice - once in the flash message, and once in the form partial. I'm asking if it's useful to display them twice?

Reply:
![screen shot 2018-10-16 at 5 48 16 pm](https://user-images.githubusercontent.com/5289943/47051906-1affc900-d15b-11e8-9e24-2ab110b5cf46.png)